woodwork/type_sys/type_system.py

```diff
@@ -391,9 +391,23 @@ def get_inference_matches(types_to_check, series, type_matches=[]):
             check_next = []
             for logical_type in types_to_check:
                 inference_func = self.inference_functions.get(logical_type)
-                if inference_func and inference_func(series):
-                    type_matches.append(logical_type)
-                    check_next.extend(self._get_children(logical_type))
+                if inference_func:
```

Comment:
I wasn't sure how much we would gain by adding a test for the special logic here. I'm leaning towards the thought that testing this would be testing implementation details and not the desired end result.